backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -3019,6 +3022,33 @@ public ExperimentRun cloneExperimentRun(CloneExperimentRun cloneExperimentRun, U
     }
 
     desExperimentRunBuilder.clearOwner().setOwner(authService.getVertaIdFromUserInfo(userInfo));
-    return insertExperimentRun(desExperimentRunBuilder.build(), userInfo);
+    return insertExperimentRun(projectDAO, desExperimentRunBuilder.build(), userInfo);
+  }
+
+  private void validateExperimentRunPerWorkspaceForTrial(
+      ProjectDAO projectDAO, String projectId, UserInfo userInfo)
+      throws InvalidProtocolBufferException, ModelDBException {
+    if (app.getTrialEnabled()) {
+      Project project = projectDAO.getProjectByID(projectId);
+      if (project.getWorkspaceId() != null && !project.getWorkspaceId().isEmpty()) {
+        WorkspaceDTO workspaceDTO =
+            roleService.getWorkspaceDTOByWorkspaceId(
+                userInfo, project.getWorkspaceId(), project.getWorkspaceTypeValue());
+        FindExperimentRuns findExperimentRuns =
+            FindExperimentRuns.newBuilder()
+                .setIdsOnly(true)
+                .setWorkspaceName(workspaceDTO.getWorkspaceName())
+                .build();
+        ExperimentRunPaginationDTO paginationDTO =
+            findExperimentRuns(projectDAO, userInfo, findExperimentRuns);
```

Comment:
add a TODO that this can be replaced by a count(*) query instead

Reply:
added todo